conan/tools/gnu/autotoolstoolchain.py

```diff
@@ -115,21 +124,46 @@ def environment(self):
             self.cxxflags.append("-fPIC")
             self.cflags.append("-fPIC")
 
+        # FIXME: Previously these flags where checked if already present at env 'CFLAGS', 'CXXFLAGS'
+        #        and 'self.cxxflags', 'self.cflags' before adding them
+        for f in list(filter(bool, [self.apple_isysroot_flag,
+                                    self.apple_arch_flag,
+                                    self.apple_min_version_flag])):
+            self.cxxflags.append(f)
+            self.cflags.append(f)
+            self.ldflags.append(f)
+
         env.append("CPPFLAGS", ["-D{}".format(d) for d in self.defines])
         env.append("CXXFLAGS", self.cxxflags)
         env.append("CFLAGS", self.cflags)
         env.append("LDFLAGS", self.ldflags)
         return env
 
-    def generate(self):
-        env = self.environment()
-        env.save_sh("conanautotoolstoolchain.sh")
-        env.save_bat("conanautotoolstoolchain.bat")
+    def generate(self, env=None):
```

Comment:
What is the purpose of this ``env=None`` environment?

Note that it will fail to override an empty environment, some users might want "do nothing", and pass an empty dict, but it will still define the default env?